src/dune/action_exec.ml

```diff
@@ -387,6 +399,40 @@ and exec_list ts ~ectx ~eenv =
     | Need_more_deps _ as need -> Fiber.return need
     | Done -> exec_list rest ~ectx ~eenv )
 
+and exec_pipe outputs ts ~ectx ~eenv =
+  let tmp_file () =
+    Path.of_string
+      (Filename.temp_file "pipe-action-"
```

Comment:
In `process.ml`, there is a `Temp` sub-module that ensures that temporary files are deleted in case of abnormal exit. We should extract it to reuse it here.